Formula/fish.rb

```diff
@@ -22,9 +22,16 @@ class Fish < Formula
 
   def install
     system "autoconf" if build.head? || build.devel?
+    args = %W[
+      --prefix=#{prefix}
+      --with-extra-functionsdir=#{HOMEBREW_PREFIX}/share/fish/vendor_functions.d
```

Comment:
It doesn't, and if the directory doesn't exist then there's no real problems, but other packaging systems would generally expect the directory to be owned by `fish`

> Does fish actually put anything in that directory at compile time, or is it purely used to teach fish to look for functions there as well as the default location?

It doesn't put anything there; it doesn't even need to exist, but the approach for other packaging systems would generally be for fish to "own" the directory. Homebrew and Nix (among others) do challenge the standard approach a bit, though.